samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -251,17 +266,20 @@ public TestRunner addOutputStream(InMemoryOutputDescriptor<?> streamDescriptor,
    * @throws SamzaException if Samza job fails with exception and returns UnsuccessfulFinish as the statuscode
    */
   public void run(Duration timeout) {
-    Preconditions.checkState(app != null,
-        "TestRunner should run for Low Level Task api or High Level Application Api");
+    Preconditions.checkState(app != null, "No Samza application to run");
     Preconditions.checkState(!timeout.isZero() || !timeout.isNegative(), "Timeouts should be positive");
+    // Cleaning store directories to ensure current run does not pick up state from previous run
+    deleteStoreDirectories();
     final LocalApplicationRunner runner = new LocalApplicationRunner(app, new MapConfig(configs));
     runner.run();
     boolean timedOut = !runner.waitForFinish(timeout);
     Assert.assertFalse("Timed out waiting for application to finish", timedOut);
```

Comment:
Throw an exception or use Preconditions instead of using junit Assert here. You can also inline this check in the block below. `if (timeout or status == unsuccessful finish) throw meaningful exception`

Reply:
sure